File: lib/core/lexicon.go
package core

import "github.com/TheOpenDictionary/odict/lib/types"

// Lexicon returns all originally-cased entries
// of a compiled dictionary
func Lexicon(dict *types.DictionaryBuffer) []string {
	words := []string{}

	for i := 0; i < dict.EntriesLength(); i++ {
		var entry types.EntryBuffer
		dict.Entries(&entry, i)
		words = append(words, string(entry.Term()))
	}

	return words
}

File: lib/src/core/lexicon.rs
use crate::{ArchivedDictionary, Dictionary};

macro_rules! lexicon {
    ($t:ident) => {
        impl $t {
            pub fn lexicon(&self) -> Vec<&str> {
                let mut vec: Vec<&str> = self
                    .entries
                    .iter()
                    .map(|(_, entry)| entry.term.as_str())
                    .collect();
                vec.sort();
                vec
            }
        }
    };
}

lexicon!(Dictionary);
lexicon!(ArchivedDictionary);
